OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -50,11 +48,19 @@ protected override void OnInside(Actor self)
 
 			var exp = self.Owner.PlayerActor.TraitOrDefault<PlayerExperience>();
 			if (exp != null)
-				exp.GiveExperience(experience);
+				exp.GiveExperience(trait.Info.PlayerExperience);
 
-			if (!string.IsNullOrEmpty(notification))
+			if (!string.IsNullOrEmpty(trait.Info.Notification))
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
-					notification, self.Owner.Faction.InternalName);
+					trait.Info.Notification, self.Owner.Faction.InternalName);
+		}
+
+		public override Activity Tick(Actor self)
+		{
+			if (trait.IsTraitDisabled)
+				return CanceledTick(self);
```

Comment:
`CanceledTick` is called from `Enter` in two places:
* If the trait has been cancelled
* If the inner activities have completed.

In other words, it is called by `Enter` when the activity knows that it is in a safe state to advance to the next activity.

By not calling the proper cancellation code you skip the piece of code that makes the spy exit the actor it is infiltrated if it has already started moving inside it.  This leaves the actor visually stuck on top of the target.